lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -822,32 +822,19 @@ ChangeResult TypeAnalyzer::visitAtenMmOp(
   auto &rhs = operands[1]->getValue();
   auto knowledge =
       ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+
+  // `aten.mm` expects both operands to be rank-2 tensors.
+  if (lhs.sizes.size() != 2 || rhs.sizes.size() != 2)
```

Comment:
I think that lhs.sizes.size() > 0 implies lhs.hasSizes in practice for our code, but you might want to mention in a comment that you are relying on that (or just check hasSizes explicitly)